src/web/src/app/space/service.ts

```diff
@@ -186,36 +198,49 @@ export class SpaceService {
 	}
 
 	join(spaceId: string, userId: string): Observable<any> {
-		return this.http.post(this._apiUrl + "/invitation", {
+		return this.apiSpaceService.handlerInvitationPost("user", {
 			spaceId: spaceId,
 			creatorId: userId,
 			userId: userId,
-			role: SpaceRole.Guest
+			role: ApiInvitationRole.RoleEnum.Guest
 		}).pipe(
 			src => this.requestWatcher.WatchFor(src)
 		);
 	}
 
 	joinByLink(token: string): Observable<any> {
-		return this.http.post(this._apiUrl + "/invitation/link?id=" + token, {})
+		return this.apiSpaceService.handlerLinkPost("user", token)
 			.pipe(
 				src => this.requestWatcher.WatchFor(src)
 			);
 	}
 
 	changeRoleInInvitation(id: number, role: string): Observable<any> {
-		return this.http.put(this._apiUrl + "/invitation?id=" + id, { role }).pipe(
+		let roleEnum = this.roleFromString(role);
+		return this.apiSpaceService.handlerInvitationPut("user", id, {role: roleEnum}).pipe(
 			src => this.requestWatcher.WatchFor(src)
 		);
 	}
 
 	approveInvitation(id: number): Observable<any> {
-		return this.http.patch(this._apiUrl + "/invitation?id=" + id, {}).pipe(
+		return this.apiSpaceService.handlerInvitationPatch("user", id).pipe(
 			src => this.requestWatcher.WatchFor(src)
 		);
 	}
 
 	resetIsChecked() {
 		this._isChecked = false;
 	}
+
+	roleFromString(role: string): ApiInvitationRole.RoleEnum {
+		// todo - а как обработать неизвестную роль? сначала добавить ее в enum на стороне openapi?
+		switch (role) {
+			case "guest":
+				return ApiInvitationRole.RoleEnum.Guest;
+			case "user":
+				return ApiInvitationRole.RoleEnum.User;
+			case "admin":
+				return ApiInvitationRole.RoleEnum.Admin;
+		}
```

Comment:
Неизвестную пока никак не обрабатываем.

Давай попробуем упростить так:
```suggestion
		return Object.keys(ApiInvitationRole.RoleEnum)
			.find(key => ApiInvitationRole.RoleEnum[key] === role);
```